api.py

```diff
@@ -0,0 +1,277 @@
+            methods=['POST'],
+        )
+        self.add_url_rule(
+            '/spiders/<int:spider_id>',
+            view_func=self.close_spider,
+            methods=['DELETE'],
+        )
+        self.add_url_rule(
+            '/database',
+            view_func=self.import_db,
+            methods=['POST'],
+        )
+        self.add_url_rule(
+            '/database',
+            view_func=self.export_db,
+            methods=['GET'],
+        )
+        self.add_url_rule(
+            '/database',
+            view_func=self.clear_scraps,
+            methods=['DELETE'],
+        )
+        self.add_url_rule(
+            '/crawls',
+            view_func=self.list_crawls,
+            methods=['GET'],
+        )
+        self.add_url_rule(
+            '/crawls',
+            view_func=self.stop,
+            methods=['DELETE'],
+        )
+        self.add_url_rule(
+            '/',
+            view_func=self.home,
+            methods=['GET'],
+        )
+
+    def home(self):
+        routes = [
+            {
+                "url": "/spiders",
+                "method": "GET"
+            },
+            {
+                "url": "/spiders",
+                "method": "POST",
+                "arguments": {
+                    "spider": "name of the spider to run"
+                }
+            },
+            {
+                "url": "/spiders/:spider_id",
+                "method": "DELETE",
+                "arguments": {
+                    "spider_id": "uuid of the spider to close"
+                }
+            },
+            {
+                "url": "/crawls",
+                "method": "GET"
+            },
+            {
+                "url": "/crawls",
+                "method": "DELETE"
+            },
+            {
+                "url": "/database",
+                "method": "GET"
+            },
+            {
+                "url": "/database",
+                "method": "POST",
+                "arguments": {
+                    "file": "json file containing the database dump"
+                }
+            },
+            {
+                "url": "/database",
+                "method": "DELETE"
+            }
+        ]
+        result = self._get_meta_response({"routes": routes})
+        return jsonify(result), 200
+
+    def list_spiders(self):
+        spiders = self.process.spider_loader.list()
+        return jsonify({"spiders": spiders, "status": "success"}), 200
+
+    def run_spider(self):
+        post_data = request.get_json()
+        spider = post_data.get('spider')
+        if spider == 'who_iris':
+            spider = who_iris_spider.WhoIrisSpider()
+        elif spider == 'nice':
+            spider = nice_spider.NiceSpider()
+        else:
+            return '', 400
+        spider_id = str(uuid.uuid4())
+        self.process.crawl(spider, uuid=spider_id)
+        crawl = self.process.join()
+        self.database.insert_spider(spider.name, spider_id)
```

Comment:
is there error handling in the database? do we need to check that the database recorded the spider successfully before carrying on?

Reply:
There is error handling in the database. We can carryon without the spider in the database. The main purpose of this is to have a stats for the spider after it finishes

Reply:
👍